docs/reference/search/request/highlighting.asciidoc

```diff
@@ -45,35 +43,43 @@ from versions before 5.0) that match the expression to be highlighted.
 Note that all other fields will not be highlighted. If you use a custom mapper and want to
 highlight on a field anyway, you have to provide the field name explicitly.
 
-[[plain-highlighter]]
-==== Plain highlighter
+[[unified-highlighter]]
+==== Unified Highlighter
 
-The default choice of highlighter is of type `plain` and uses the Lucene highlighter.
-It tries hard to reflect the query matching logic in terms of understanding word importance and any word positioning criteria in phrase queries.
+The default choice of highlighter is of type `unified` and uses the Lucene Unified highlighter.
 
-[WARNING]
-If you want to highlight a lot of fields in a lot of documents with complex queries this highlighter will not be fast.
-In its efforts to accurately reflect query logic it creates a tiny in-memory index and re-runs the original query criteria through
-Lucene's query execution planner to get access to low-level match information on the current document.
-This is repeated for every field and every document that needs highlighting. If this presents a performance issue in your system consider using an alternative highlighter.
+This highlighter breaks the text into sentences and scores individual sentences as
+if they were documents in this corpus, using the BM25 algorithm.
+It also supports accurate phrase and multi-term (fuzzy, prefix, regex) highlighting.
+
+[float]
+===== Offsets Strategy
```

Comment:
Offsets Strategy requires some explanation, perhaps:

In order to create meaningful search snippets from the terms being queried, a highlighter needs to know the start and end character offsets of each word in the original text.  These offsets can be obtained from:

* The postings list (fields mapped as `"index_options": "offsets"`).
* Term vectors (fields mapped as `"term_vectors": "with_positions_offsets"`).
* The original field, by reanalysing the text on-the-fly.